From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: l1b0k <libokang.dev@gmail.com>
Date: Mon, 23 May 2022 17:43:50 +0800
Subject: [PATCH] terway: support kubelet health check

when package travel cross netns fw mark will lost ,this will cause bpf unable to identify host network.

Signed-off-by: l1b0k <libokang.dev@gmail.com>
---
 bpf/bpf_lxc.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/bpf/bpf_lxc.c b/bpf/bpf_lxc.c
index 6192691586..64f9169958 100644
--- a/bpf/bpf_lxc.c
+++ b/bpf/bpf_lxc.c
@@ -1634,8 +1634,7 @@ int tail_ipv6_to_endpoint(struct __ctx_buff *ctx)
 				 * as the host. So we can ignore the ipcache
 				 * if it reports the source as HOST_ID.
 				 */
-				if (sec_label != HOST_ID)
-					src_identity = sec_label;
+				src_identity = sec_label;
 			}
 		}
 		cilium_dbg(ctx, info ? DBG_IP_ID_MAP_SUCCEED6 : DBG_IP_ID_MAP_FAILED6,
@@ -1976,8 +1975,7 @@ int tail_ipv4_to_endpoint(struct __ctx_buff *ctx)
 				 * as the host. So we can ignore the ipcache
 				 * if it reports the source as HOST_ID.
 				 */
-				if (sec_label != HOST_ID)
-					src_identity = sec_label;
+				src_identity = sec_label;
 			}
 		}
 		cilium_dbg(ctx, info ? DBG_IP_ID_MAP_SUCCEED4 : DBG_IP_ID_MAP_FAILED4,
-- 
2.39.1

